File: Sources/Indigox.SSO/Services/RegisteredService.cs
using System;
using Indigox.SSO.Interface;

namespace Indigox.SSO.Services
{
    public class RegisteredService : IRegisteredService, IEquatable<IRegisteredService>
    {
        private int id;
        private string serviceID;
        private string name;
        private string secretKey;
        private string accessLoginUrl;
        private string loginUrl;
        private string loginOutUrl;
        private DateTime createTime;
        private bool isEnabled;
        private bool isWindowsAuthentication;
        private bool isDefaultService;
        private bool isAllowedToProxy;

        public int ID
        {
            get { return id; }
            set { id = value; }
        }

        public string ServiceID
        {
            get { return serviceID; }
            set { serviceID = value; }
        }

        public string Name
        {
            get { return name; }
            set { name = value; }
        }

        public string SecretKey
        {
            get { return secretKey; }
            set { secretKey = value; }
        }

        public string AccessLoginUrl
        {
            get { return accessLoginUrl; }
            set { accessLoginUrl = value; }
        }

        public string LoginUrl
        {
            get { return loginUrl; }
            set { loginUrl = value; }
        }

        public string LoginOutUrl
        {
            get { return loginOutUrl; }
            set { loginOutUrl = value; }
        }

        public DateTime CreateTime
        {
            get { return createTime; }
            set { createTime = value; }
        }

        public bool IsEnabled
        {
            get { return isEnabled; }
            set { isEnabled = value; }
        }

        public bool IsWindowsAuthentication
        {
            get { return isWindowsAuthentication; }
            set { isWindowsAuthentication = value; }
        }

        public bool IsDefaultService
        {
            get { return isDefaultService; }
            set { isDefaultService = value; }
        }

        public bool IsAllowedToProxy
        {
            get { return isAllowedToProxy; }
            set { isAllowedToProxy = value; }
        }

        public bool Equals(IRegisteredService other)
        {
            return this.serviceID == other.ServiceID;
        }
    }
}

File: Sources/Indigox.SSO.Application/js/Indigox/SSO/Application/RegisteredService.js
define("Indigox/SSO/Application/RegisteredService",
    [
        "Indigox.Web.JsLib.Models.RecordManager"
    ],
function (
    RecordManager
) {
    RecordManager.getInstance().register('RegisteredService', {
        columns: [
            { name: 'ID', text: '编号', type: String },
            { name: 'ServiceID', text: '系统编号', type: String },
            { name: 'Name', text: '系统名称', type: String },
            { name: 'SecretKey', text: '密钥', type: String },
            { name: 'AccessLoginUrl', text: 'WinAuth链接', type: String },
            { name: 'LoginUrl', text: '登录链接', type: String },
            { name: 'LoginOutUrl', text: '注销链接', type: String },
            { name: 'IsEnabled', text: '是否可用', type: String },
            { name: 'IsDefaultService', text: '是否默认系统', type: String },
            { name: 'IsWindowsAuthentication', text: '是否为Windows验证', type: String },
            { name: 'IsAllowedToProxy', text: '是否允许代理', type: String },
            { name: "CreateTime", text: '创建时间', type: String }
        ],
        primaryKey: ['ID']
    });

});